imap_processing/swe/l1a/swe_science.py

```diff
@@ -0,0 +1,168 @@
+    it collect data for 12 energy steps and at each energy step,
+    it collects 7 data from each 7 CEMs.
+
+    Each L1A data from each packet will have this shape: 15 rows, 12 columns,
+    and each cell in 15 x 12 table contains 7 element array.
+    These dimension maps to this:
+        15 rows --> 15 seconds
+        12 column --> 12 energy steps in each second
+        7 element --> 7 CEMs counts
+
+    In L1A, we don't do anything besides read raw data, uncompress counts data and
+    store data in 15 x 12 x 7 array.
+
+    SWE want to keep all value as it is in L1A. Post L1A, we group data into full cycle
+    and convert raw data to engineering data as needed.
+
+    Parameters
+    ----------
+    packet_file : str
+        packet file path
+
+    Returns
+    -------
+    xarray.Dataset
+        xarray dataset with data.
+    """
+    science_array = []
+
+    metadata_arrays = collections.defaultdict(list)
+
+    # We know we can only have 8 bit numbers input, so iterate over all
+    # possibilities once up front
+    decompression_table = np.array([uncompress_counts(i) for i in range(256)])
+
+    for data_packet in decom_data:
+        # read raw data
+        binary_data = data_packet.data["SCIENCE_DATA"].raw_value
+        # read binary string to an int and then convert it to
+        # bytes. This is to convert the string to bytes.
+        # Eg. "0000000011110011" --> b'\x00\xf3'
+        byte_data = int(binary_data, 2).to_bytes(1260, byteorder="big")
+        # convert bytes to numpy array of uint8
+        raw_counts = np.frombuffer(byte_data, dtype=np.uint8)
+
+        # Uncompress counts. Uncompressed data is a list of 1260
+        # where 1260 = 15 seconds x 12 energy steps x 7 CEMs
+        # Take the "raw_counts" indices/counts mapping from
+        # decompression_table and then reshape the return
+        uncompress_data = np.take(decompression_table, raw_counts).reshape(15, 12, 7)
+
+        # Save data with its metadata field to attrs and DataArray of xarray.
+        science_array.append(uncompress_data)
+        add_metadata_to_array(data_packet, metadata_arrays)
+
+    science_xarray = xr.DataArray(
+        science_array,
+        dims=["number_of_packets", "seconds", "energy_steps", "cem_counts"],
+    )
+
+    met_time = xr.DataArray(
+        metadata_arrays["SHCOARSE"],
+        name="MET",
+        dims=["number_of_packets"],
+        attrs=dict(
+            description="Mission elapsed time",
+            units="seconds since start of the mission",
+        ),
+    )
+
+    dataset = xr.Dataset(
+        {"SCIENCE_DATA": science_xarray},
+        coords={"met_time": met_time},
+    )
+
+    # create xarray dataset for each metadata field
+    for key, value in metadata_arrays.items():
+        if key == "SHCOARSE":
+            continue
+
+        dataset[key] = xr.DataArray(
+            value,
+            dims=["number_of_packets"],
+        )
```

Comment:
@bryan-harter will it look something like this when we add time to be the first dimension.
```suggestion
    met_time = xr.DataArray(
        metadata_arrays["SHCOARSE"],
        name="met_time",
        dims=["met_time"],
        attrs=dict(
            description="Mission elapsed time",
            units="seconds since start of the mission",
        ),
    )

    science_xarray = xr.DataArray(
        science_array,
        dims=["met_time", "seconds", "energy_steps", "cem_counts"],
    )

    dataset = xr.Dataset(
        {"SCIENCE_DATA": science_xarray},
        coords={"met_time": met_time},
    )

    # create xarray dataset for each metadata field
    for key, value in metadata_arrays.items():
        if key == "SHCOARSE":
            continue

        dataset[key] = xr.DataArray(
            value,
            dims=["met_time"],
        )
```